repose-aggregator/components/filters/regex-rbac-filter/src/test/scala/org/openrepose/filters/regexrbac/RegexRbacFilterTest.scala

```diff
@@ -0,0 +1,504 @@
+
+      When("the protected resource is requested")
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      Then(s"the request should not be allowed access")
+      val delegationValue = filterChain.getRequest.asInstanceOf[HttpServletRequest].getHeader(HttpDelegationHeaderNames.Delegated)
+      delegationValue should not be null
+      val delegationHeader = parseDelegationHeader(delegationValue)
+      delegationHeader shouldBe a[Success[_]]
+      delegationHeader.get.statusCode shouldBe SC_FORBIDDEN
+      delegationHeader.get.component shouldBe componentName
+      delegationHeader.get.quality shouldBe quality
+    }
+    Seq("/path/To/it", "/path/to/it").foreach { path =>
+      Seq(
+        // @formatter:off
+        //Role          Result
+        ("role1,role2", SC_OK),
+        ("role1"      , SC_FORBIDDEN),
+        ("role2"      , SC_FORBIDDEN)
+        // @formatter:on
+      ).foreach { case (roles, result) =>
+        it(s"${shouldNot(result)} allow the request to $path with roles '$roles'.") {
+          Given(s"a request to $path with roles '$roles'")
+          servletRequest.setRequestURI(path)
+          servletRequest.setMethod("GET")
+          servletRequest.addHeader(XRolesHeader, roles)
+          val resources = new ResourcesType
+          resources.setValue(
+            """/path/[^/]+/.*     get role1
+              |/path/[Tt][^/]+/.* GET role2
+              |""".stripMargin.trim())
+          config.setResources(resources)
+          filter.configurationUpdated(config)
+
+          When("the protected resource is requested")
+          filter.doFilter(servletRequest, servletResponse, filterChain)
+
+          Then(s"the request should not be allowed access")
+          servletResponse.getStatus shouldBe result
+          if (result == SC_OK) {
+            assert(filterChain.getRequest.asInstanceOf[HttpServletRequest].getHeader(XRelevantRolesHeader).contains("role1"))
+            assert(filterChain.getRequest.asInstanceOf[HttpServletRequest].getHeader(XRelevantRolesHeader).contains("role2"))
+          } else {
+            assert(filterChain.getRequest == null)
+          }
+        }
+      }
+    }
+  }
+
+  describe("when destroying the filter") {
+    it("should unregister the configuration from the configuration service") {
+      Given("an un-initialized filter and a mock'd Filter Config")
+      filter.configuration shouldBe null
+      !filter.isInitialized
+      mockFilterConfig.addInitParameter("filter-config", "another-name.cfg.xml")
+
+      When("the filter is initialized and destroyed")
+      filter.init(mockFilterConfig)
+      filter.destroy()
+
+      Then("the filter should unregister with the ConfigurationService")
+      verify(mockConfigService).unsubscribeFrom(eql("another-name.cfg.xml"), same(filter))
+    }
+  }
+}
+
+object RegexRbacFilterTest {
+  private final val XRolesHeader = "X-Roles"
+  private final val XRelevantRolesHeader = "X-Relevant-Roles"
```

Comment:
Since this is a thing now, might as well make it a constant as well

Reply:
Updated.